python/ray/data/tests/test_dataset.py

```diff
@@ -2997,13 +2997,42 @@ def test_groupby_arrow_sum(ray_start_regular_shared, num_parts):
     assert [row.as_pydict() for row in agg_ds.sort("A").iter_rows()] == \
         [{"A": 0, "sum(B)": 1683}, {"A": 1, "sum(B)": 1617},
          {"A": 2, "sum(B)": 1650}]
+
```

Comment:
Can we add test cases for the all-None cases? I think this is where the behavior is less defined.